napalm/nxos_ssh/nxos_ssh.py

```diff
@@ -1008,6 +1008,73 @@ def cli(self, commands):
             cli_output[py23_compat.text_type(command)] = output
         return cli_output
 
+    def get_environment(self):
+        """
+        Get environment facts.
+
+        power and fan are currently not implemented
+        cpu is using 1-minute average
+        """
+
+        environment = {}
+        # sys_resources contains cpu and mem output
+        sys_resources = self._send_command('show system resources')
+        temp_cmd = 'show env temp'
+
+        # cpu
+        environment.setdefault('cpu', {})
+        environment['cpu'][0] = {}
+        environment['cpu'][0]['%usage'] = 0.0
+        for line in sys_resources.splitlines():
+            # Load average:   1 minute: 0.14   5 minutes: 0.18   15 minutes: 0.21
+            match = re.search(r'1 minute: [0-9]+(.[0-9]+)?', line)
+            if match is not None:
+                cpu = match.group(0).split(':')[1]
+                cpu = '{0:.2f}'.format(float(cpu))
+                environment['cpu'][0]['%usage'] = float(cpu)
+                break
+        else:
```

Comment:
This should be indented in. Just to clarify the goal, it is that if you see a line you are not expecting, error out.

Reply:
The full output of show system resources is:
```
nx01-top# show system resources
Load average:   1 minute: 0.05   5 minutes: 0.28   15 minutes: 0.27
Processes   :   642 total, 1 running
CPU states  :   1.02% user,   2.57% kernel,   96.40% idle
        CPU0 states  :   1.03% user,   4.12% kernel,   94.84% idle
        CPU1 states  :   1.05% user,   2.10% kernel,   96.84% idle
        CPU2 states  :   1.02% user,   1.02% kernel,   97.95% idle
        CPU3 states  :   2.04% user,   2.04% kernel,   95.91% idle
Memory usage:   16401224K total,   5005012K used,   11396212K free
Current memory status: OK
```
I'm not expecting the load average to be the first line, knowing Cisco I'm allowing the flexibility that it could be on a different line on a different NX-OS. So this loops through the whole output looking for the regex and if found breaks out of the loop, if the regex isn't found on any line then ValueError is raised.

Reply:
I'll defer to @ktbyers but my opinion and experience is if you don't know the data you will get more false positives. Obviously doing it the way you intend would less likely to fail, but again, more likely to have "needle in the haystack" type issues.

Reply:
I'll try using the textfsm extractor instead